contribs/gmf/src/directives/themeselector.js

```diff
@@ -87,18 +101,14 @@ gmf.ThemeselectorController = function($scope, ngeoLocation, gmfThemes) {
    */
   this.filter;
 
-  /**
-   * @type {ngeo.Location}
-   * @private
-   */
-  this.ngeoLocation_ = ngeoLocation;
-
-  /**
-   * @type {gmf.Themes}
-   * @private
-   */
-  this.gmfThemes_ = gmfThemes;
-
+  $scope.$watchCollection(function() {
+    return this.currentTheme;
+  }.bind(this), function(newTheme, oldTheme) {
+    if (newTheme && newTheme !== oldTheme &&
+        this.gmfTreeManager_.isModeFlush()) {
+      this.setLocationPath_(newTheme['name']);
```

Comment:
Now the "currentTheme" is the theme in the TreeManager. This theme (the object) "never change". Only its properties change. So I should use a `watchCollection`.